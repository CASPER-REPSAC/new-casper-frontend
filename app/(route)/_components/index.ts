'use client';

export { default as Background } from './Background';
export { default as NoticeSection } from './NoticeSection';
export { default as TitleSection } from './TitleSection';
