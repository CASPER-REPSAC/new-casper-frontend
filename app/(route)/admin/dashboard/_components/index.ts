'use client';

export { default as FileLogTable } from './FileLogTable';
export { default as CommentLogTable } from './CommentLogTable';
export { default as PostLogTable } from './PostLogTable';
export { default as LoginLogTable } from './LoginLogTable';
