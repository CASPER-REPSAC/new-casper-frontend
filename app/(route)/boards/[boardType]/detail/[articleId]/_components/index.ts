export { default as DetailComment } from './DetailComment';
export { default as AuthorSection } from './AuthorSection';
export { default as ContentSection } from './ContentSection';
export { default as ButtonSection } from './ButtonSection';
export { default as CommentEditorSection } from './CommentEditorSection';
