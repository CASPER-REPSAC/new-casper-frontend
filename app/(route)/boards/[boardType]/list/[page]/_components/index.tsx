export { default as BoardHeader } from './BoardHeader';
export { default as BoardFooter } from './BoardFooter';
export { default as BoardBody } from './BoardBody';
export { default as PostLink } from './PostLink';
