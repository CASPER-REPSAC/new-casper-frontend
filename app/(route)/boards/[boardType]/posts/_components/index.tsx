'use client';

export { default as EditorSection } from './EditorSection';
export { default as TitleSection } from './TitleSection';
export { default as WriteButtonSection } from './WriteButtonSection';
export { default as BoardTypeSelecSection } from './BoardTypeSelecSection';
export { default as PostFormProvider } from './PostFormProvider';
export { default as OptionSection } from './OptionSection';
