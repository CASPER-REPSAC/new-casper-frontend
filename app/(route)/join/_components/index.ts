'use client';

export { default as JoinFunnel } from './JoinFunnel';
export { default as PageWrapper } from './PageWrapper';
