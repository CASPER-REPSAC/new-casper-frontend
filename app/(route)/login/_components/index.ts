'use client';

export { default as Logo } from './Logo';
export { default as LoginForm } from './LoginForm';
export { default as LoginMoreLinks } from './LoginMoreLinks';
