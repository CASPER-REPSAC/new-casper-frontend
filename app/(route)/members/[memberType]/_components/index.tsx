'use client';

export { default as DetailMemberCard } from './DetailMemberCard';
export { default as MemberCard } from './MemberCard';
export { default as PageTemplate } from './PageTemplate';
