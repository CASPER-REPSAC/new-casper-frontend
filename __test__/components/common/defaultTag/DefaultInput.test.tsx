import { DefaultInput } from '@app/_components/common';
import { fireEvent, render, screen } from '@testing-library/react';

describe('<DefaultInput />', () => {
  test('snapshot test', () => {
    const { container } = render(
      <DefaultInput
        placeholder="값을 입력하세요."
        defaultValue="default value"
      />,
    );

    expect(container).toMatchSnapshot();
  });

  test('정상적으로 값이 입력된다.', () => {
    render(<DefaultInput placeholder="값을 입력하세요." />);

    const input = screen.getByPlaceholderText('값을 입력하세요.');
    fireEvent.change(input, { target: { value: 'value changed' } });

    expect(input).toHaveValue('value changed');
  });
});
