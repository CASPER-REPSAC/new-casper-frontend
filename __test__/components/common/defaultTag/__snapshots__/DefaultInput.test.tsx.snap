// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DefaultInput /> snapshot test 1`] = `
<div>
  <input
    class="input  "
    placeholder="값을 입력하세요."
    value="default value"
  />
</div>
`;
